spec/rubocop/cop/performance/sample_spec.rb

```diff
@@ -5,118 +5,74 @@
-      new_source = autocorrect_source(cop, '[1, 2, 3, 4].shuffle[0..3]')
-
-      expect(new_source).to eq('[1, 2, 3, 4].shuffle[0..3]')
+    it "when using #{acceptable} on a collection variable" do
+      inspect_source(cop, ['coll = [1, 2, 3]', "coll.#{acceptable}"].join("\n"))
+      expect(cop.messages).to be_empty
     end
+  end
 
-    it 'does not correct shuffle with an exclusive range selector' do
-      new_source = autocorrect_source(cop, '[1, 2, 3, 4].shuffle[0...3]')
-
-      expect(new_source).to eq('[1, 2, 3, 4].shuffle[0...3]')
-    end
-
-    it 'corrects shuffle with an array range selector' do
-      new_source = autocorrect_source(cop, '[1, 2, 3, 4].shuffle[0, 3]')
-
-      expect(new_source).to eq('[1, 2, 3, 4].sample(3)')
+  shared_examples 'corrects' do |wrong, right|
+    it "corrects #{wrong} to #{right}" do
+      new_source = autocorrect_source(cop, "[1, 2, 3].#{wrong}")
+      expect(new_source).to eq("[1, 2, 3].#{right}")
     end
+  end
 
-    it 'corrects shuffle with an assigned random hash' do
-      source = '[1, 2, 3, 4].shuffle(random: Random.new(1))'
-      new_source = autocorrect_source(cop, source)
-
-      expect(new_source).to eq('[1, 2, 3, 4].sample(random: Random.new(1))')
+  shared_examples 'does not correct' do |acceptable|
+    it "does not correct #{acceptable}" do
+      new_source = autocorrect_source(cop, "[1, 2, 3].#{acceptable}")
+      expect(new_source).to eq("[1, 2, 3].#{acceptable}")
     end
+  end
 
-    it 'corrects shuffle with an assigned random variable' do
-      source = ['random = { random: Random.new(1) }',
-                '[1, 2, 3, 4].shuffle(random)'].join("\n")
-      new_source = autocorrect_source(cop, source)
+  fixes = {
+    'shuffle.first'  => 'sample',
+    'shuffle.last'   => 'sample',
+    'shuffle[0]'     => 'sample',
+    'shuffle[2]'     => 'sample',
+    'shuffle[0, 3]'  => 'sample(3)',
+    'shuffle[0..3]'  => 'sample(4)',
+    'shuffle[0...3]' => 'sample(3)',
+    # 'shuffle(random: Random.new).first' => 'sample(random: Random.new)',
+    # 'shuffle(random: Random.new)[2]'    => 'sample(random: Random.new)',
+    # 'shuffle(random: Random.new)[0, 3]' => 'sample(3, random: Random.new)',
+    # 'shuffle(random: Random.new)[0..3]' => 'sample(4, random: Random.new)',
+  }
+
+  fixes.each do |wrong, right|
+    it_behaves_like('registers offense', wrong, right)
+    it_behaves_like('corrects',          wrong, right)
+  end
 
-      expect(new_source).to eq(['random = { random: Random.new(1) }',
-                                '[1, 2, 3, 4].sample(random)'].join("\n"))
-    end
+  acceptables = [
+    'sample',
+    'shuffle',
+    'shuffle(random: Random.new)',
+    'shuffle.join([5, 6, 7])',
+    'shuffle.map { |e| e }',
+    'shuffle(random: Random.new).find(&:odd?)',
+  ]
+
+  acceptables.each do |acceptable|
+    it_behaves_like('does not register offense', acceptable)
+    it_behaves_like('does not correct',          acceptable)
```

Comment:
Consider inlining the array here. Personally, one of the benefits that I see with `it_behaves_like` is the ability to run that individual test group for a single `acceptable`. With this setup, a group has to be run with all `acceptables`. The only disadvantage, that I see, is code duplication.